### PR TITLE
Normalize the block toolbar of the audio, file, media and text, video, site logo and post featured image blocks.

### PR DESCRIPTION
**Pull request by youknowriad**
Related #25983 similar to #29205 #29247 and #29863

Similar to the previous PRs, this one uses the new BlockControls "groups" (block, inline, other) for the following blocks:

 - Audio
 - File
 - List
 - Video
 - Media & Text
 - Site Logo
 - Post Featured Image

In order to do so, I had to remove the "ToolbarGroup" that was explicitly added to some reusable components (toolbars).

**Testing instructions**

 - Check that the block toolbar of the blocks above look and work properly.

### CODE REVIEW

File: packages/block-library/src/media-text/edit.js [hunk 1]
```diff
@@ -187,20 +187,6 @@ function MediaTextEdit( { attributes, isSelected, setAttributes } ) {
 		gridTemplateColumns,
 		msGridColumns: gridTemplateColumns,
 	};
-	const toolbarControls = [
```
Comment (gziolo) [inline packages/block-library/src/media-text/edit.js:190]: Should we discourage the usage of `controls` prop at the end of the refactoring with all non-default groups in `BlockControls `?

Comment (youknowriad) [reply packages/block-library/src/media-text/edit.js:190]: I think it's not supported in non default group blocks (I did this on purpose as I think component composition is better)

**Review by gziolo — APPROVED** [review]
I did a quick sanity check for the code changes and it looks similar to previous PRs. Everything looks good in my opinion. It's far important to test the impact on the UI.

**Comment by jasmussen** [comment]
Nice work, I'm so excited for you to get to the navigation block soon :D 

These are all excellent:

Screenshot 2021-03-19 at 11 22 35

Screenshot 2021-03-19 at 11 23 15

Screenshot 2021-03-19 at 11 23 26

_though it would be nice to revisit dropdowns for list style and indentation_ ;) 

Screenshot 2021-03-19 at 11 23 48

Screenshot 2021-03-19 at 11 24 56

I'm seeing a small issue with the site logo:

Screenshot 2021-03-19 at 11 24 36

Is that because I'm inserting it in the post editor when I should be inserting it in the site editor?

For media & text. it helps with a "before":

Screenshot 2021-03-19 at 11 27 30

There was so much going on there it was hard to see what was what. So that aspect is definitely improved:

Screenshot 2021-03-19 at 11 24 11

With this drastic reduction, some things I need to work on separately become clear:

- Left/right icons aren't properly centered
- The media & Text icon itself could use an upgrade

I think we should put Justification next to Alignment, as those two feel related, then left/right can come after.

That _might_ also improve the appearance of the link icon, which right now feels like it just sits there on its own.

**Comment by gziolo** [comment]
Nice teamwork @jasmussen with reviewing, it's very effective 💯 

I love all the small refinements that improve the readability of the block toolbar.

**Comment by youknowriad** [comment]
@jasmussen For the site logo block, it seems that you're in a context where "left/right/center" alignments don't seem to be allowed. I'm not sure how I can do that (in post editor, I see them personally), 

That said, this highlighted something for me. we have a `alignWide` flag in some blocks that is used to remove the "wide/full" alignments from some blocks. The site logo block use that, and for me this flag doesn't really make sense at all. Let me explain: even if the site logo doesn't expand its width when "fully aligned", it still make sense to "aligned fully" which basically means aligned it by the edge of its container and not inside the "content size" area. This is a thought for now, but I might try to remove that flag entirely separately.

**Comment by jasmussen** [comment]
Yes, that's an excellent thought. I think it touches on the wide/fullwide alignment discussion we've had, where I see it as a property of post content or other containers optionally. In most cases, like the site logo, I would imagine its position be largely dictated by its parent, rather than on its own. Give or take.

**Comment by youknowriad** [comment]
>  In most cases, like the site logo, I would imagine its position be largely dictated by its parent, rather than on its own. Give or take.

I'm not sure I agree with this personally, I think all blocks should have all the alignments options that their parent allow them to have. This means that in a regular context, all blocks should be allowed to be aligned left/right/wide/full.

In some context, parent blocks could remove some things from their children, examples:

 - Column block remove "wide/full" from its children
 - Buttons block remove all alignments options from its children.

--- 
Also, this is not an adhoc thing, if the container is another kind of layout (think flex, grid, absolute), all these alignments options should be removed from all children  and replaced by new ones.

**Comment by youknowriad** [comment]
In other words, I don't think it's the responsibility of a single block to decide what alignments options it can have.

**Comment by jasmussen** [comment]
I was unclear, and I was thinking primarily of how the Group block centers content inside by default. I would not want you to have to make a Header template part be "full-wide" or else it was in a centered column. I definitely agree you should be able to left align a site logo inside such a block (though I can see a situation where we have flex properties on a group, and space-between plus margins might be the better aligner). 

I need to take a weekend, I think, thanks again for this PR!

**Pull request merged**